future-apps/payroll/contracts/Payroll.sol

```diff
@@ -250,121 +324,159 @@ contract Payroll is AragonApp { //, IForwarder { // makes coverage crash (remove
     /**
      * @dev Set token distribution for payments to an employee (the caller)
      * @notice Set token distribution for payments to an employee (the caller).
-     *         Only callable once every 6 months
      * @param tokens Array with the tokens to receive, they must belong to allowed tokens for employee
-     * @param distribution Array (correlated to tokens) with the proportions (integers over 100)
+     * @param distribution Array (correlated to tokens) with the proportions (integers summing to 100)
      */
-    function determineAllocation(address[] tokens, uint8[] distribution) external {
+    function determineAllocation(address[] tokens, uint8[] distribution) isInitialized employeeMatches external {
         Employee storage employee = employees[employeeIds[msg.sender]];
-        // check that employee exists (and matches)
-        require(employee.accountAddress == msg.sender);
 
         // check arrays match
         require(tokens.length == distribution.length);
 
+        // delete previous allocation
+        for (uint32 j = 0; j < allowedTokensArray.length; j++) {
+            delete employee.allocation[allowedTokensArray[j]];
+        }
+
         // check distribution is right
         uint8 sum = 0;
-        uint32 i;
-        for (i = 0; i < distribution.length; i++) {
+        for (uint32 i = 0; i < distribution.length; i++) {
             // check token is allowed
             require(allowedTokens[tokens[i]]);
             // set distribution
             employee.allocation[tokens[i]] = distribution[i];
-            sum += distribution[i];
-            require(sum >= distribution[i]);
+            sum = sum.add(distribution[i]);
         }
         require(sum == 100);
+
+        emit DetermineAllocation(employeeIds[msg.sender], msg.sender);
     }
 
     /**
      * @dev To withdraw payment by employee (the caller). The amount owed since last call will be transferred.
      * @notice To withdraw payment by employee (the caller). The amount owed since last call will be transferred.
      */
-    function payday() external {
+    function payday() isInitialized employeeMatches external {
```

Comment:
I think `isInitialized` is not needed here, as if `employeeMatches` is passing it means there's an element in `employees` with `accountAddress` non-zero, with can only be done after adding an employee and therefore having had the contract initialized. I would remove it and add a comment explaining it. Same applies to other functions with both modifiers.